Malmo/samples/Python_examples/robust_frames.py

```diff
@@ -0,0 +1,319 @@
+
+# -- set up the agent host --
+agent_host = MalmoPython.AgentHost()
+try:
+    agent_host.parse( sys.argv )
+except RuntimeError as e:
+    print 'ERROR:',e
+    print agent_host.getUsage()
+    exit(1)
+if agent_host.receivedArgument("help"):
+    print agent_host.getUsage()
+    exit(0)
+
+# -- set up the mission --
+xml = '''<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
+<Mission xmlns="http://ProjectMalmo.microsoft.com" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://ProjectMalmo.microsoft.com Mission.xsd">
+  <About>
+    <Summary/>
+  </About>
+  <ModSettings>
+    <MsPerTick>1</MsPerTick>
+  </ModSettings>
+  <ServerSection>
+    <ServerHandlers>
+      <FlatWorldGenerator forceReset="false" generatorString="3;7,220*1,5*3,2;3;,biome_1" seed=""/>
+      <ServerQuitFromTimeUp description="" timeLimitMs="130000"/>
+      <ServerQuitWhenAnyAgentFinishes description=""/>
+    </ServerHandlers>
+  </ServerSection>
+  <AgentSection mode="Survival">
+    <Name>Cristina</Name>
+    <AgentStart/>
+    <AgentHandlers>
+      <ObservationFromFullStats/>
+      <VideoProducer viewpoint="0" want_depth="false">
+        <Width>320</Width>
+        <Height>240</Height>
+      </VideoProducer>
+      <RewardForReachingPosition dimension="0">
+        <Marker oneshot="true" reward="100" tolerance="1.1" x="19.5" y="0" z="19.5"/>
+      </RewardForReachingPosition>
+    </AgentHandlers>
+  </AgentSection>
+</Mission>'''
+my_mission = MalmoPython.MissionSpec(xml,True)
+
+# -- test each action set in turn --
+max_retries = 3
+for action_set in ['discrete_absolute','discrete_relative', 'teleport']:
+
+    if action_set == 'discrete_absolute':
+        my_mission.allowAllDiscreteMovementCommands()
+    elif action_set == 'discrete_relative':
+        my_mission.allowAllDiscreteMovementCommands()
+    elif action_set == 'teleport':
+        my_mission.allowAllAbsoluteMovementCommands()
+    else:
+        print 'ERROR: Unsupported action set:',action_set
+        exit(1)
+
+    for retry in range(max_retries):
+        try:
+            agent_host.startMission( my_mission, MalmoPython.MissionRecordSpec() )
+            break
+        except RuntimeError as e:
+            if retry == max_retries - 1:
+                print "Error starting mission:",e
+                exit(1)
+            else:
+                time.sleep(2.5)
+
+    print "Waiting for the mission to start",
+    world_state = agent_host.getWorldState()
+    while not world_state.has_mission_begun:
+        sys.stdout.write(".")
+        time.sleep(0.1)
+        world_state = agent_host.getWorldState()
+        for error in world_state.errors:
+            print "Error:",error.text
+    print
+
+    # the main loop:
+    agent = RandomAgent( agent_host, action_set )
+    world_state = agent.waitForInitialState()
+    while world_state.is_mission_running:
+        agent.act()
+        world_state = agent.waitForNextState()
```

Comment:
Would be very interesting to get some timings from this. How long does one act/wait iteration take on average, and how does it vary depending on action set?